src/project.cc

```diff
@@ -120,8 +120,9 @@ const std::vector<std::string>& GetSystemIncludes(
 
   // Capture these flags in |extra_flags|, since they may change system include
   // directories.
-  static std::vector<std::string> kFlagsToPass = {"--gcc-toolchain",
-                                                  "--sysroot"};
+  static std::vector<std::string> kFlagsToPass = {
+      "--gcc-toolchain", "--sysroot",     "-nostdinc",
+      "-nostdinc++",     "-nobuiltininc", "-stdlib=libstdc++"};
```

Comment:
Should this be `-stdlib=` instead of `-stdlib=libstdc++`?

Reply:
There's more to it than that.
The current algorithm for `--gcc-toolchain` and `--sysroot` is:
- If the full flag is `--sysroot=something` take the full flag and be done.
- If the full flag is `--sysroot` take the flag *and the next one* because we need the associated path.

The `-no*` family of flags doesn't have a path associated, so matching the full flag will result in undesired behaviour\*.

The `-stdlib=<lib>` is again special.
- If we put it in `kFlagsToPass` as `-stdlib`, the full flag found would either be `-stdlib=libc++` or `-stdlib=libstdc++` - either way, the current algorithm works, but kind of by accident.
- If we put it in `kFlagsToPass` as `-stdlib=<lib>`, the full flag will be matched and the current algorithm would take the next flag as well, assuming there should be a path associated.

Reply:
Apologies, I forgot to explain "undesired behaviour".

If it blindly picks the next flag (assuming this is called before sanitisation), the flag might be absolutely anything - even something that will cause an AST deserialization error.
If it is called after the flags have been sanitized, it still might take a bad flag. Imagine pulling in `-I` without the associated path.